src/ninshiki_cpp_detector.cpp

```diff
@@ -87,35 +81,48 @@ int main(int argc, char ** argv)
           if (value == 0 || value == 1) {
             myriad = value;
           } else {
-            std::cout << "Unknown option for MYRIAD `" << arg << "`!\n\n" <<
-              help_message << std::endl;
+            std::cout << "Unknown option for MYRIAD `" << arg << "`!\n\n"
+                      << help_message << std::endl;
             return 1;
           }
         } else if (arg == "--frequency") {
           frequency = atoi(argv[i++]);
+        } else if (arg == "--method") {
         } else {
           std::cout << "Unknown argument `" << arg << "`!\n\n" << help_message << std::endl;
           return 1;
         }
-      } else if (pos == 0) {
+      } else if (arg == "dnn") {
+        dnn_detector = std::make_shared<ninshiki_cpp::detector::DnnDetector>();
+      } else if (arg == "color") {
+        color_detector = std::make_shared<ninshiki_cpp::detector::ColorDetector>();
+      } else if (arg == "lbp") {
+        lbp_detector = std::make_shared<ninshiki_cpp::detector::LBPDetector>();
```

Comment:
Seriously you should utilize [argparse](https://github.com/p-ranav/argparse), but don't do that here, just do that on a separate PR when you have the time in the future.